In [1]:
%matplotlib inline

# PLANE OF SKY (POS) MAGNETIC FIELD STRENGTH I
Author: Jordan Guerra (for Villanova University). December 2023.

The purpose of this tutorial is to illustrate the use of the python package *polBpy* for calculating the POS magnetic field strength using different Davis-Chandrasekhar-Fermi (DCF) approximations.

This tutorial uses data from literature listed [here.](https://github.com/jorgueagui/polBpy/blob/9039d4af5d25c49130bf51be7fe0ce363424edcc/refs.md)

**EXAMPLE I:** This example shows the most basic use of *polBpy*, calculating $B_{POS}$ using the classical DCF approximation. We reproduce some results from Chuss et. al. (2019), which presents results of OMC-1 using observations of SOFIA/HAWC+.

In [2]:
import os
os.chdir(os.path.expanduser("/Users/jguerraa/Desktop/polBpy"))
from polBpy import DCF

Let us calculate $B_{POS}$ for the BN/KL region (Figure 8), using the 53 $\mu$m data. Values necessary for this calcuation can found in (and around) Table 5 of Chuss+19. 

In [3]:
col_den = 9.85E+22 # cm^-2
vel_disp = 1.85 # km/s
cloud_depth = 4.34E+17 # cm
ang_disp = (0.37)**0.5 # Root-square of <B_t^2>/<B_0^2>

In [4]:
bpos = DCF.dcf_classical(col_den,vel_disp,ang_disp,rho=False,cdepth=cloud_depth)

The option rho=False (default) is used since the density value is colunm density.

In [5]:
print(bpos) # Result is in micro G

(1111.9107639357849, 0.0)


Comment: the resulting $B_{POS}$ is slightly larger than that reported in Chuss+19 because (most likely) the used value of $\mu$(mean molecular weight) was lower than the one implemented in *polBpy* ($\mu$=2.8).

**EXAMPLE II:** Now we reproduce the calcuations from Lopez-Rodriguez+21. In this example we can use the built-in option for calculating the associated uncertainty for $B_{POS}$.

First, using the classical approximation, and the simple (e.g. $\sigma$ value of the PA distribution) angle dispersion. In this case, a correction factor ($\eta$=0.5; Ostriker+2001) can be used,

In [6]:
col_den = 1.78E-22 #g cm^-3
vel_disp = 66.0 #km s^-1
ang_disp = 0.29 # rad
u_col_den = 0.45E-22
u_vel_disp = 6.6 # km s^-1
eta = 0.5 #

In [7]:
bpos = DCF.dcf_classical(col_den,vel_disp,ang_disp,rho=True,eta=eta,uden=u_col_den,uvel=u_vel_disp)

In this case the option rho=True is used since density is true mass density, and no depth of the cloud is needed.

In [8]:
print(bpos) # In micro G

(538.1841151602063, 86.74301945129082)


This result is interpreted as $B_{POS}=538\pm87$ $\mu$G.

Now, let us calculate $B_{POS}$ by using the large-scale flow modified DCF. Using the same values above, we specified the flow values,

In [9]:
ls_flow = 396.
u_ls_flow = 87.

In [10]:
bpos = DCF.dcf_ls_flow(col_den,vel_disp,ang_disp,ls_flow,rho=True,eta=eta,uden=u_col_den,uvel=u_vel_disp,uflow=u_ls_flow)

In [11]:
print(bpos) # In micro G

(398.2562452185525, 218.53326571534362)


A different $B_{POS}$ can be calculated using the same DCF approximations but now using an angle dispersion value from the dispersion-function analysis. In this case, no correction factor is needed ($\eta=1.0$ - default value).

In [12]:
ang_disp = (0.07)**0.5 # Root-square of <B_t^2>/<B_0^2>
u_ang_disp = (0.01)**0.5

In [13]:
bpos = DCF.dcf_classical(col_den,vel_disp,ang_disp,rho=True,uden=u_col_den,uvel=u_vel_disp,udisp=u_ang_disp)

In [14]:
print(bpos) # In micro G

(1179.8039577170946, 484.776319632109)


Now using the large-scale flow modification

In [15]:
bpos = DCF.dcf_ls_flow(col_den,vel_disp,ang_disp,ls_flow,rho=True,uden=u_col_den,uvel=u_vel_disp,uflow=u_ls_flow)

In [16]:
print(bpos) # In micro G

(693.0767630404232, 436.9198872903982)


Comments: